### PR TITLE
(DEBUG) S4 Mk3: compare scratch functions `engine.scratch..` vs `engine.setValue(group, "scratch2..")`

### PR DESCRIPTION
**Pull request by ronso0**
Helper to fix/debug #14000 

This adds two options for scratching:
* 3 scratch modes:
   curr: `engine.setValue(group, "scratch2..)`
   alt1: `engine.scratchEnable()`/`engine.scratchTick()` 
   alt2: `scratch_position_enable` / `scratch_position` (currently used for waveform scratching)
* sync scratch speed with wheel LED speed (~2x as fast as the current scratch speed)

With the alternative methods (scratchTick, wheelPos) scratching sounds pretty smooth **at low speeds**, while with the current implementation it's rather just noise (no buffer underruns, really just how wheel ticks are filtered & emitted)

**Another notable difference:**
with the `scratchTick()` method the track is very slowly drifting backwards when doing many (40+) fast + short back & forth scratches¹.

### My results:
**curr: `scratch2`:** 
* noisy at low speeds
* small drift for fast back'n'forth scratches¹
* no notable drift for one-direction scratches
(some revs forth, then same backwards, should land at the same position)

**alt1: `scratchTick`:**
* smooth sound at all speeds
* drifts backwards  for fast back'n'forth scratches¹
* no notable drift for one-direction scratches

**alt2: `scratchPos`:**
* bit noisy at very low speeds
* **no drift** whatsoever!

### Bottom line:
So, to get the best of both methods, could we use the alt. method for very slow scratches?
~~I'll test that soon.~~
**edit** No, I won't. I'm using `scratchPos` and it suits my needs. Sounds good a mid/high speeds, no drift.

Or fix the filtering for the current method?
I have no clue how..

### CODE REVIEW

File: res/controllers/Traktor-Kontrol-S4-MK3.js [hunk 1]
```diff
@@ -2413,8 +2423,16 @@ class S4Mk3Deck extends Deck {
...
+                        if (useEngineScratch) {
+                            engine.scratchEnable(this.deck.currentDeckNumber,
+                                ScratchIntervalsPerRev,
+                                BaseRevolutionsPerMinute,
+                                ScratchAlpha,
+                                ScratchBeta); // ramoing true by default
+                        } else {
+                            engine.setValue(this.group, "scratch2_enable", true);
+                        }
+                    } else { // release
```
Comment (acolombier) [inline res/controllers/Traktor-Kontrol-S4-MK3.js:2435]: This looks wrong - seems like `engine.setValue(this.group, "scratch2_enable", true);` is always run both in `if` and `else` if `useEngineScratch` isn't used

**Review by acolombier — COMMENTED** [review]
I haven't had a chance to test yet, but I'll try to do that ASAP

File: res/controllers/Traktor-Kontrol-S4-MK3.js (hunk 1 above)
Comment (ronso0) [reply res/controllers/Traktor-Kontrol-S4-MK3.js:2435]: I don't understand, sorry. The diff looks good, I just added an alternative path for every enable/disable branch.
But indeed, with `engine.scratchEnable()` `scratch2_enable` is also set true (in c++).

Just give a spin (haha!) when you find some time and let me know how it works for you.

**Review by acolombier — CHANGES_REQUESTED** [review]
The new scratching experience feels very good and precision change doesn't feel as bad as I feared. I'm happy to get those two settings as part of the core mapping. :+1: 

Only "regression" from my perceptive is the hotcue not "cutting" the scratch anymore. 
My routine being, I would scratch over, say a voice sampler and would be able to resume playing instantly by activating a hotcue. How would you feel about restoring that behaviour via an alternative setting? (e.g `Stop scratch when activating hotcue` and slight refactor of L2458-2470)

Another bug is, if you set a jogwheel tension higher than `0.80`, the jog will speed up and never stop! I guess you might be able to adjust the range.

Let me know if you would want me to suggest patches

**Comment by ronso0** [comment]
> Only "regression" from my perceptive is the hotcue not "cutting" the scratch anymore.
How would you feel about restoring that behaviour via an alternative setting? (e.g Stop scratch when activating hotcue and slight refactor of L2458-2470)

Sure, I don't plan to remove features, though I'd make the cutting opt-in.

**Comment by ronso0** [comment]
> The new scratching experience feels very good and precision change doesn't feel as bad as I feared. I'm happy to get those two settings as part of the core mapping. 👍

Great to hear!
So we'd have 2 new scratching options then, both opt-in:
* make scratch speed match wheel LED speed
* seeking to hotcue stops scratching

Or where you saying the engine.scratch.. functions should be optional, too?
Given that both this and scratch2 show sticker drift and engine.scratch.. sounds much smoother at low speeds I'd actually drop scratch2.
Wdyt?

**Comment by ronso0** [comment]
Bit weird: I've been using PositionScratchController lately and it works smoothly, but it seems I get let's say variable results. Fast scratching, eg. when drop-cueing a track, sometimes sounds rather harsh. Not sure where that stems from.
Will revert to scratchTick() until I figured what's wrong. Bit of a pity since PositionScratchController sounds much better at slow rates.

**Comment by ywwg** [comment]
Is this something we should look at merging officially for making the MK3 feel better?  Even as a selectable option

**Comment by ronso0** [comment]
Yup, IMO that would make sense.
I'll test again, but as far as I remember the scratch feeling was better. Especially as it eliminates the drift entirely.

Did you test it @ywwg ?

Dunno when I have the nerves to rebase and test, but the conflict is probably only because of the new "stop scratch on hotcue press" option.
